File: C#/Escuela/Segundo parcial/Kardex escuela/Kardex.cs
using System;
namespace Kardex
{
    public class NewProgram
    {
        public static void Main(string[] args)
        {
            string nombreDeLaEscuela, nombreDelAlumno, gradoDelAlumno, grupoDelAlumno, periodoDelAlumno, mes;
            int edadDelAlumno;
            float calificacionEspanol, calificacionMatematicas, calificacionIngles, calificacionQuimica, calificacionFisica;

            while (true)
            {
                Console.WriteLine("Introduce tus datos. \n");
                try
                {
                    Console.Write("Ingresa el nombre de tu escuela: ");
                    nombreDeLaEscuela = Console.ReadLine();

                    Console.Write("Ingresa tu nombre completo: ");
                    nombreDelAlumno = Console.ReadLine();

                    Console.Write("Ingresa tu edad: ");
                    edadDelAlumno = int.Parse(Console.ReadLine());

                    Console.Write("Ingresa tu grado: ");
                    gradoDelAlumno = Console.ReadLine();

                    Console.Write("Ingresa tu grupo: ");
                    grupoDelAlumno = Console.ReadLine();

                    Console.Write("Ingresa tu periodo, el NÚMERO de semestre: ");
                    periodoDelAlumno = Console.ReadLine();

                    Console.Write("En qué mes estás: ");
                    mes = Console.ReadLine();

                    // Calificaciones
                    Console.Write("Ingresa tu calificación de español del uno al diez: ");
                    calificacionEspanol = float.Parse(Console.ReadLine());

                    Console.Write("Ingresa tu calificación de matemáticas del uno al diez: ");
                    calificacionMatematicas = float.Parse(Console.ReadLine());

                    Console.Write("Ingresa tu calificación de inglés del uno al diez: ");
                    calificacionIngles = float.Parse(Console.ReadLine());

                    Console.Write("Ingresa tu calificación de química del uno al diez: ");
                    calificacionQuimica = float.Parse(Console.ReadLine());

                    Console.Write("Ingresa tu calificación de física del uno al diez: ");
                    calificacionFisica = float.Parse(Console.ReadLine());

                    // Se rompe el bucle si todos los datos se ingresan correctamente
                    break;
                }
                catch (Exception e)
                {
                    Console.WriteLine("Algo salió mal, no metiste tus datos correctamente, inténtalo de nuevo");
                    Console.WriteLine($"Tipo de error: {e}");
                }
            }

            float promedio = (calificacionEspanol + calificacionMatematicas + calificacionFisica + calificacionQuimica + calificacionIngles) / 5;
            string estatusDelEstudiante = promedio >= 6 ? "Aprobado" :  "Reprobado";

            Console.WriteLine($"\n---------------------\nNombre del alumno: {nombreDelAlumno}\nEdad: {edadDelAlumno}\nGrado: {gradoDelAlumno}\nGrupo: {grupoDelAlumno}\nPeriodo {periodoDelAlumno}\nMes actual: {mes}\n\n\nEspañol: {calificacionEspanol}\nMatemáticas: {calificacionMatematicas}\nInglés: {calificacionIngles}\nQuímica: {calificacionQuimica}\nFísica: {calificacionFisica}\n\nPromedio: {promedio}\nEstado: {estatusDelEstudiante}");
        }
    }
}

File: C#/Escuela/Segundo parcial/Kardex escuela/Kardex.py
# El c# está muy feo de trabajar, voy a usar python, si no le molesta
import sys


def main(*args: None) -> int:
    while True:
        print("Introduce tus datos.\n")
        try:
            nombre_de_la_escuela = input("Ingresa el nombre de tu escuela: ")
            nombre_del_alumno = input("Ingresa tu nombre completo: ")
            edad_del_alumno = int(input("Ingresa tu edad: "))
            grado_del_alumno = input("Ingresa tu grado: ")
            grupo_del_alumno = input("Ingresa tu grupo: ")
            periodo_del_alumno = input("Ingresa tu periodo, el NÚMERO de semestre: ")
            mes = input("En qué mes estás: ")

            materias = ("español", "matematicas", "ingles", "quimica", "fisica")
            calificaciones = {}
            for materia in materias:
                calificaciones[materia] = float(
                    input(f"Ingresa tu calificación de {materia} del uno al diez: ")
                )

            promedio = sum(calificaciones.values()) / len(materias)

            estatus_del_estudiante: str = "Aprobado" if promedio >= 6 else "Reprobado"

            print(
                f"\n---------------------\nNombre del alumno: {nombre_del_alumno}\nEdad:{edad_del_alumno}\nGrado: {grado_del_alumno}\nGrupo: {grupo_del_alumno}\nPeriodo {periodo_del_alumno}\nMes actual: {mes}\n\n\nEstado: {estatus_del_estudiante}"
            )

            for materia, calificacion in calificaciones.items():
                print(f"{materia.capitalize()}: {calificacion}")

            print(f"\nPromedio: {promedio}\nEstado: {estatus_del_estudiante}")

            return 0

        except ValueError as e:
            print(
                "Algo salió mal, no metiste tus datos correctamente, inténtalo de nuevo"
            )
            print(f"Tipo de error: {e}")


if __name__ == "__main__":
    sys.exit(main())
